### PR TITLE
Implemented KMP Algorithm in Python

### PR DESCRIPTION
**Pull request by anubhvshrma18**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description

Algorithm used for searching a pattern/string in a another text/string using an array of prefixes and suffixes. It overrides the Naive algorithm and improves the worst case to O(n).

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

 #427

### CODE REVIEW

**Comment by anubhvshrma18** [comment]
Please review @HarshCasper  @Kajol-Kumari

File: Python/cp/KMP_Algorithm.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+text = "aaanunaanuaunnanuubhanu"
+pattern = "anu"
+KMPSearch(pattern, text)
+
+
+# ''' Output
+# Pattern found at 3
+# Pattern found at 8
+# Pattern found at 15
+# Pattern found at 21
```
Comment (Kajol-Kumari) [inline Python/cp/KMP_Algorithm.py]: please add one extra blank line here.

File: Python/cp/KMP_Algorithm.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            else:
+                i += 1
+
+#for dynamic input use the below code 
+# text =input()
+# pattern = input()
+# KMPSearch(pattern, text)
+
+
+text = "aaanunaanuaunnanuubhanu"
```
Comment (Kajol-Kumari) [inline Python/cp/KMP_Algorithm.py]: please comment these static input. i.e line 52 to 54.

File: Python/cp/KMP_Algorithm.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            print("Pattern found at " + str(i - j + 1)) # print the pattern starting index
+            j = lps[j - 1]
+        elif i < sizeText and pattern[j] != text[i]:
+            if j != 0:
+                j = lps[j - 1]
+            else:
+                i += 1
+
+#for dynamic input use the below code 
+# text =input()
```
Comment (Kajol-Kumari) [inline Python/cp/KMP_Algorithm.py]: please uncomment lines 47-49 and show a message like `please enter the input string` and `please enter the string to be searched` respectively in the both input statements.

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
Hey @anubhvshrma18 the `readme.md` is showing many unnecessary lines change, if possible please revert back the `readme.md` changes and just modify the line where you need to add you code path.
Functionality is working fine! Please incorporated the suggested changes and it will be goofd to go :)

**Comment by anubhvshrma18** [comment]
@Kajol-Kumari Updated all the changes as per requested . Please review and if anything more is to be changed please remind me, i will do that too.

**Review by Kajol-Kumari — CHANGES_REQUESTED** [review]
Hey @anubhvshrma18 please resolve the conflicts. And i can see many changes in readme can u please reset head only for `readme.md` and then add your code path in there? 

You can use `git checkout origin/master readme.md` for undoing the `readme.md` changes

**Review by vatsalkesarwani12 — APPROVED** [review]
LGTM 👍

**Review by Kajol-Kumari — APPROVED** [review]
LGTM :)

**Comment by iamrajiv** [comment]
@all-contributors please add @anubhvshrma18 for code

**Pull request merged**